src/patterns/ec2-app/base.ts

```diff
@@ -158,176 +194,11 @@ function restrictedCidrRanges(ranges: IPeer[]) {
- * for example the number of instances in an ASG.
- *
- * To do this, we can use a simple ternary:
- * ```typescript
- * new GuEc2App(this, {
- *   // other required properties
- *   scaling: this.stage === "PROD" ? { minimumInstances: 6 } : { minimumInstances: 1 }
- * });
- * ```
- *
- * If multiple properties need to be stage aware, consider encoding this into the `props` of the calling class:
- *
- * ```typescript
- * export interface MyAppProps extends GuStackProps {
- *   domainName: string;
- *   scaling: GuAsgCapacity;
- *   monitoringConfiguration: Alarms | NoMonitoring;
- * }
- *
- * export class MyApp extends GuStack {
- *   private static app = "cdk-playground";
- *
- *   constructor(scope: App, id: string, props: MyAppProps) {
- *     const { stage, domainName, scaling, monitoringConfiguration } = props;
- *
- *     super(scope, id, props);
- *
- *     new GuPlayApp(this, {
- *       app: "my-app",
- *       instanceType: InstanceType.of(InstanceClass.T4G, InstanceSize.MICRO),
- *       access: { scope: AccessScope.PUBLIC },
- *       userData: {
- *         distributable: {
- *           fileName: `my-app.deb`,
- *           executionStatement: `dpkg -i /my-app/my-app.deb`,
- *         },
- *       },
- *       certificateProps: {
- *         domainName,
- *       },
- *       monitoringConfiguration,
- *       scaling,
- *     });
- *   }
- * }
- *
- * new MyApp(app, "MyApp-CODE", {
- *   stage: "CODE",
- *   domainName: "my-app.code.dev-gutools.co.uk",
- *   scaling: { minimumInstances: 1 },
- *   monitoringConfiguration: { noMonitoring: true },
- * });
- *
- * new MyApp(app, "MyApp-PROD", {
- *   stage: "PROD",
- *   domainName: "my-app.prod.gutools.co.uk",
- *   scaling: { minimumInstances: 6 },
- *   monitoringConfiguration: {
- *     // config to enable alarms in PROD
- *   },
- * });
- * ```
- *
- * If you _must_ support a non-standard VPC setup, it's possible to override the VPC and subnet selection into which
- * instances will be started. You need to set the three props below in order for this to work:
- *  vpc: this must be an IVpc instance indicating the VPC to deploy into (usually obtained from `Vpc.fromVpcAttributes`)
- *  privateSubnets: a list of ISubnet indicating the subnets for the autoscaling group to deploy instances into (usually
- *    obtained from `GuVpc.subnets`)
- *  publicSubnets: a list of ISubnet indicating the subnets for the loadbalancer to use, if the loadbalancer
- *    is intended for public access (usually obtained from `GuVpc.subnets`)
- *    On the other hand, if the access is private, then an "internal" loadbalancer is used in the `privateSubnets` list.
- * Please note that no checking is applied to the subnet lists to ensure that they correspond to the vpc given; as
- * an advanced feature it is assumed that the caller ensures this.  If the subnets are mismatched then the resulting
- * Cloudformation will fail to deploy.
- *
- * ### Customising the infrastructure generated by this pattern
```

Comment:
Again, this is useful information, but more general than this pattern, so I wonder if we should lift it elsewhere?

Reply:
(i.e. we wouldn't want to repeat it on all of our patterns - it's more some general info on how to use CDK.)